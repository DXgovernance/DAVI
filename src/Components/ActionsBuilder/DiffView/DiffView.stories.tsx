import { ComponentStory, ComponentMeta } from '@storybook/react';
import { DiffView } from './DiffView';

export default {
  title: 'Components/DiffView',
  component: DiffView,
} as ComponentMeta<typeof DiffView>;

function DemoDiffView() {
  const oldCode = `
  {
    "colors": {
      "text": "red",
      "primary": "#295FF4",
      "secondary": "#181B1B",
      "red": "#E75C5C",
      "orange": "#D17A06",
      "proposalText": {
        "lightGrey": "#BDC0C7",
        "grey": "orange"
      },
    }
  }
`;
  const newCode = `
  {
    "colors": {
      "text": "#fff",
      "primary": "#295FF4",
      "secondary": "#181B1B",
      "red": "#E75C5C",
      "orange": "#D17A06",
      "proposalText": {
        "grey": "#A1A6B0",
        "lightGrey": "#BDC0C7"
      },
    }
  }
`;

  return <DiffView oldCode={oldCode} newCode={newCode} />;
}

const Template: ComponentStory<typeof DiffView> = () => <DemoDiffView />;

export const Simple = Template.bind({});
Simple.storyName = 'JSON';
