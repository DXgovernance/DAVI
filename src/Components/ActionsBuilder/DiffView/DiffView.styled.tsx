import styled from 'styled-components';

export const DiffViewCodeFoldMessage = styled.span`
  font-weight: 500;
`;

export const DiffCode = styled.pre`
  line-height: 20px;
`;
