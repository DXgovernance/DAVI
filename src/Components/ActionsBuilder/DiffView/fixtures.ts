export const mockCodeOriginal = `
  {
    "colors": {
      "text": "red",
      "primary": "#295FF4",
      "secondary": "#181B1B",
      "red": "#E75C5C",
      "orange": "#D17A06",
      "proposalText": {
        "lightGrey": "#BDC0C7",
        "grey": "orange"
      },
    }
  }
`;

export const mockCodeUpdated = `
  {
    "colors": {
      "text": "#fff",
      "primary": "#295FF4",
      "secondary": "#181B1B",
      "red": "#E75C5C",
      "orange": "#D17A06",
      "proposalText": {
        "grey": "#A1A6B0",
        "lightGrey": "#BDC0C7"
      },
    }
  }
`;
