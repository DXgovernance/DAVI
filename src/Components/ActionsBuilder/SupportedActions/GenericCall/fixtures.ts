import {
  ApproveSendTokens,
  DecodedCall,
  SupportedAction,
} from 'Components/ActionsBuilder/types';
import ERC20ABI from 'contracts/ERC20.json';
import { BigNumber, utils } from 'ethers';

const ERC20Contract = new utils.Interface(ERC20ABI.abi);

export const genericDecodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.GENERIC_CALL,
  function: ERC20Contract.getFunction('increaseAllowance'),
  to: '0x3f943f38b2fbe1ee5daf0516cecfe4e0f8734351',
  value: BigNumber.from(0),
  args: {
    param1: BigNumber.from(10000),
    param2: BigNumber.from(10000),
    param3: '0x3f943f38b2fbe1ee5daf0516cecfe4e0f8734351',
  },
};

export const approvetokenSpendingMock: ApproveSendTokens = {
  amount: BigNumber.from(1000),
  token: '0x3f943f38b2fbe1ee5daf0516cecfe4e0f8734351',
};
