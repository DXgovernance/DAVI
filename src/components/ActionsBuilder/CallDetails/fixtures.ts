import {
  ApproveSendTokens,
  DecodedCall,
  SupportedAction,
} from 'components/ActionsBuilder/types';
import ERC20 from 'contracts/ERC20.json';
import { BigNumber, utils } from 'ethers';

const ERC20Contract = new utils.Interface(ERC20.abi);

export const emptyDecodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.ERC20_TRANSFER,
  function: ERC20Contract.getFunction('transfer'),
  to: '',
  value: BigNumber.from(0),
  args: {
    _to: '',
    _value: BigNumber.from(0),
  },
};

export const decodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.ERC20_TRANSFER,
  function: ERC20Contract.getFunction('transfer'),
  to: '0x95a223299319022a842d0dfe4851c145a2f615b9',
  value: BigNumber.from(0),
  args: {
    _to: '0x0000000000000000000000000000000000000000',
    _value: BigNumber.from(1000),
  },
};

export const approvetokenSpendingMock: ApproveSendTokens = {
  callType: SupportedAction.GENERIC_CALL,
  from: '0x0000000000000000000000000000000000000000',
  to: '0x0000000000000000000000000000000000000000',
  value: BigNumber.from(0),
  function: null,
  args: {},
  amount: BigNumber.from(1000),
  token: '0x3f943f38b2fbe1ee5daf0516cecfe4e0f8734351',
};
