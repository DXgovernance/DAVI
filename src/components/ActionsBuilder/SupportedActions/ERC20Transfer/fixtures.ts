import { DecodedCall, SupportedAction } from 'components/ActionsBuilder/types';
import ERC20ABI from 'contracts/ERC20.json';
import { BigNumber, utils } from 'ethers';

const ERC20Contract = new utils.Interface(ERC20ABI.abi);

export const erc20TransferEmptyDecodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.ERC20_TRANSFER,
  function: ERC20Contract.getFunction('transfer'),
  to: '',
  value: BigNumber.from(0),
  args: {
    _to: '',
    _value: BigNumber.from(0),
  },
};

export const erc20TransferDecodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.ERC20_TRANSFER,
  function: ERC20Contract.getFunction('transfer'),
  to: '0xfFb1cd0F95368DDd06D556161c5D3d9f0f4Fe6d2',
  value: BigNumber.from(0),
  args: {
    _to: '0xaF8eB8C3A5d9d900AA0B98e3Df0bcC17d3C5F698',
    _value: BigNumber.from('5000000000000000000'),
  },
};
