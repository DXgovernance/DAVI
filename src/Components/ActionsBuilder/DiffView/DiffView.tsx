import ReactDiffViewer, { DiffMethod } from 'react-diff-viewer';
import hljs from 'highlight.js';
import json from 'highlight.js/lib/languages/json';
import { useTheme } from 'styled-components';
import { DiffCode, DiffViewCodeFoldMessage } from './DiffView.styled';
import './syntax.css';

hljs.registerLanguage('json', json);

export interface DiffViewProps {
  oldCode: string;
  newCode: string;
}

export const DiffView: React.FC<DiffViewProps> = ({ oldCode, newCode }) => {
  const theme = useTheme();

  const highlightSyntax = (str: string) => (
    <DiffCode
      dangerouslySetInnerHTML={{
        __html: hljs.highlight(str, { language: 'json' }).value,
      }}
    />
  );

  return (
    <ReactDiffViewer
      oldValue={oldCode}
      newValue={newCode}
      splitView={false}
      extraLinesSurroundingDiff={2}
      disableWordDiff={false}
      compareMethod={DiffMethod.WORDS}
      showDiffOnly={true}
      useDarkTheme={true}
      styles={{
        variables: {
          dark: {
            diffViewerBackground: 'transparent',
            gutterBackground: 'transparent',
            gutterBackgroundDark: 'transparent',
            gutterColor: theme.colors.proposalText.grey,
            codeFoldGutterBackground: theme.colors.syntax.expBlue,
            codeFoldBackground: theme.colors.syntax.expLight,
            addedBackground: theme.colors.syntax.addLight,
            wordAddedBackground: theme.colors.syntax.addText,
            addedGutterBackground: theme.colors.syntax.add,
            addedGutterColor: theme.colors.text,
            removedBackground: theme.colors.syntax.removeLight,
            wordRemovedBackground: theme.colors.syntax.removeText,
            removedGutterBackground: theme.colors.syntax.remove,
            removedGutterColor: theme.colors.text,
            codeFoldContentColor: theme.colors.proposalText.grey,
          },
        },
      }}
      codeFoldMessageRenderer={totalFoldedLines => (
        <DiffViewCodeFoldMessage>
          Expand {totalFoldedLines} lines of code
        </DiffViewCodeFoldMessage>
      )}
      renderContent={highlightSyntax}
    />
  );
};
