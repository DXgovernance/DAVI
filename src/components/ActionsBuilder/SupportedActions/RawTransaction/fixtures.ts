import { DecodedCall, SupportedAction } from 'components/ActionsBuilder/types';
import { BigNumber } from 'ethers';
import { MOCK_ADDRESS } from 'hooks/Guilds/ens/fixtures';

export const rawDataCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.RAW_TRANSACTION,
  function: null,
  to: MOCK_ADDRESS,
  value: BigNumber.from('1000000000000000000'),
  args: null,
  optionalProps: {
    data: '0x01',
  },
};
