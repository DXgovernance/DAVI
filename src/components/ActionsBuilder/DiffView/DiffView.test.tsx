import { render } from 'utils/tests';
import { DiffView } from './DiffView';
import { mockCodeOriginal, mockCodeUpdated } from './fixtures';

describe('DiffView', () => {
  it('Should match snapshot', () => {
    const { container } = render(
      <DiffView oldCode={mockCodeOriginal} newCode={mockCodeUpdated} />
    );
    expect(container).toMatchSnapshot();
  });
});
