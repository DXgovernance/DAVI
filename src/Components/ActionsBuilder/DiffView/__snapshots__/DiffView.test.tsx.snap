// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiffView Should match snapshot 1`] = `
<div>
   
  <table
    class="css-1vc8h0l-diff-container"
  >
    <tbody>
      <tr
        class="css-1n7ec1i-line"
      >
        <td
          class="css-jncl07-gutter"
        >
          <pre
            class="css-na97mt-line-number"
          >
            1
          </pre>
        </td>
        <td
          class="css-jncl07-gutter"
        >
          <pre
            class="css-na97mt-line-number"
          >
            1
          </pre>
        </td>
        <td
          class="css-1tuzevz-marker"
        >
          <pre />
        </td>
        <td
          class="css-vl0irh-content"
        >
          <pre
            class="css-69s43g-content-text"
          >
            <pre
              class="sc-gsTEea bovuND"
            >
                
              <span
                class="hljs-punctuation"
              >
                {
              </span>
            </pre>
          </pre>
        </td>
      </tr>
      <tr
        class="css-1n7ec1i-line"
      >
        <td
          class="css-jncl07-gutter"
        >
          <pre
            class="css-na97mt-line-number"
          >
            2
          </pre>
        </td>
        <td
          class="css-jncl07-gutter"
        >
          <pre
            class="css-na97mt-line-number"
          >
            2
          </pre>
        </td>
        <td
          class="css-1tuzevz-marker"
        >
          <pre />
        </td>
        <td
          class="css-vl0irh-content"
        >
          <pre
            class="css-69s43g-content-text"
          >
            <pre
              class="sc-gsTEea bovuND"
            >
                  
              <span
                class="hljs-attr"
              >
                "colors"
              </span>
              <span
                class="hljs-punctuation"
              >
                :
              </span>
               
              <span
                class="hljs-punctuation"
              >
                {
              </span>
            </pre>
          </pre>
        </td>
      </tr>
      <tr
        class="css-1n7ec1i-line"
      >
        <td
          class="css-jncl07-gutter css-jtsnpx-diff-removed"
        >
          <pre
            class="css-na97mt-line-number"
          >
            3
          </pre>
        </td>
        <td
          class="css-jncl07-gutter css-10oj6k6-empty-gutter css-jtsnpx-diff-removed"
        >
          <pre
            class="css-na97mt-line-number"
          />
        </td>
        <td
          class="css-1tuzevz-marker css-jtsnpx-diff-removed"
        >
          <pre>
            -
          </pre>
        </td>
        <td
          class="css-vl0irh-content css-jtsnpx-diff-removed"
        >
          <pre
            class="css-69s43g-content-text"
          >
            <span
              class="css-cncyp1-word-diff"
            >
              <pre
                class="sc-gsTEea bovuND"
              >
                      
                <span
                  class="hljs-attr"
                >
                  "text"
                </span>
                <span
                  class="hljs-punctuation"
                >
                  :
                </span>
                 
                <span
                  class="hljs-string"
                >
                  "
                </span>
              </pre>
            </span>
            <span
              class="css-cncyp1-word-diff css-1uzas5h-word-removed"
            >
              <pre
                class="sc-gsTEea bovuND"
              >
                red
              </pre>
            </span>
            <span
              class="css-cncyp1-word-diff"
            >
              <pre
                class="sc-gsTEea bovuND"
              >
                <span
                  class="hljs-string"
                >
                  ",
                </span>
              </pre>
            </span>
          </pre>
        </td>
      </tr>
      <tr
        class="css-1n7ec1i-line"
      >
        <td
          class="css-jncl07-gutter css-10oj6k6-empty-gutter css-5x3b4c-diff-added"
        >
          <pre
            class="css-na97mt-line-number"
          />
        </td>
        <td
          class="css-jncl07-gutter css-5x3b4c-diff-added"
        >
          <pre
            class="css-na97mt-line-number"
          >
            3
          </pre>
        </td>
        <td
          class="css-1tuzevz-marker css-5x3b4c-diff-added"
        >
          <pre>
            +
          </pre>
        </td>
        <td
          class="css-vl0irh-content css-5x3b4c-diff-added"
        >
          <pre
            class="css-69s43g-content-text"
          >
            <span
              class="css-cncyp1-word-diff"
            >
              <pre
                class="sc-gsTEea bovuND"
              >
                      
                <span
                  class="hljs-attr"
                >
                  "text"
                </span>
                <span
                  class="hljs-punctuation"
                >
                  :
                </span>
                 
                <span
                  class="hljs-string"
                >
                  "
                </span>
              </pre>
            </span>
            <span
              class="css-cncyp1-word-diff css-fjxrmw-word-added"
            >
              <pre
                class="sc-gsTEea bovuND"
              >
                #fff
              </pre>
            </span>
            <span
              class="css-cncyp1-word-diff"
            >
              <pre
                class="sc-gsTEea bovuND"
              >
                <span
                  class="hljs-string"
                >
                  ",
                </span>
              </pre>
            </span>
          </pre>
        </td>
      </tr>
      <tr
        class="css-1n7ec1i-line"
      >
        <td
          class="css-jncl07-gutter"
        >
          <pre
            class="css-na97mt-line-number"
          >
            4
          </pre>
        </td>
        <td
          class="css-jncl07-gutter"
        >
          <pre
            class="css-na97mt-line-number"
          >
            4
          </pre>
        </td>
        <td
          class="css-1tuzevz-marker"
        >
          <pre />
        </td>
        <td
          class="css-vl0irh-content"
        >
          <pre
            class="css-69s43g-content-text"
          >
            <pre
              class="sc-gsTEea bovuND"
            >
                    
              <span
                class="hljs-attr"
              >
                "primary"
              </span>
              <span
                class="hljs-punctuation"
              >
                :
              </span>
               
              <span
                class="hljs-string"
              >
                "#295FF4"
              </span>
              <span
                class="hljs-punctuation"
              >
                ,
              </span>
            </pre>
          </pre>
        </td>
      </tr>
      <tr
        class="css-1n7ec1i-line"
      >
        <td
          class="css-jncl07-gutter"
        >
          <pre
            class="css-na97mt-line-number"
          >
            5
          </pre>
        </td>
        <td
          class="css-jncl07-gutter"
        >
          <pre
            class="css-na97mt-line-number"
          >
            5
          </pre>
        </td>
        <td
          class="css-1tuzevz-marker"
        >
          <pre />
        </td>
        <td
          class="css-vl0irh-content"
        >
          <pre
            class="css-69s43g-content-text"
          >
            <pre
              class="sc-gsTEea bovuND"
            >
                    
              <span
                class="hljs-attr"
              >
                "secondary"
              </span>
              <span
                class="hljs-punctuation"
              >
                :
              </span>
               
              <span
                class="hljs-string"
              >
                "#181B1B"
              </span>
              <span
                class="hljs-punctuation"
              >
                ,
              </span>
            </pre>
          </pre>
        </td>
      </tr>
      <tr
        class="css-r9xyk9-code-fold"
      >
        <td
          class="css-k44qg2-code-fold-gutter"
        />
        <td
          class="css-k44qg2-code-fold-gutter"
        />
        <td />
        <td>
          <a
            tabindex="0"
          >
            <span
              class="sc-bdfBQB gXofpO"
            >
              expandLinesOfCode
            </span>
          </a>
        </td>
        <td />
        <td />
      </tr>
      <tr
        class="css-1n7ec1i-line"
      >
        <td
          class="css-jncl07-gutter"
        >
          <pre
            class="css-na97mt-line-number"
          >
            8
          </pre>
        </td>
        <td
          class="css-jncl07-gutter"
        >
          <pre
            class="css-na97mt-line-number"
          >
            8
          </pre>
        </td>
        <td
          class="css-1tuzevz-marker"
        >
          <pre />
        </td>
        <td
          class="css-vl0irh-content"
        >
          <pre
            class="css-69s43g-content-text"
          >
            <pre
              class="sc-gsTEea bovuND"
            >
                    
              <span
                class="hljs-attr"
              >
                "proposalText"
              </span>
              <span
                class="hljs-punctuation"
              >
                :
              </span>
               
              <span
                class="hljs-punctuation"
              >
                {
              </span>
            </pre>
          </pre>
        </td>
      </tr>
      <tr
        class="css-1n7ec1i-line"
      >
        <td
          class="css-jncl07-gutter css-jtsnpx-diff-removed"
        >
          <pre
            class="css-na97mt-line-number"
          >
            9
          </pre>
        </td>
        <td
          class="css-jncl07-gutter css-10oj6k6-empty-gutter css-jtsnpx-diff-removed"
        >
          <pre
            class="css-na97mt-line-number"
          />
        </td>
        <td
          class="css-1tuzevz-marker css-jtsnpx-diff-removed"
        >
          <pre>
            -
          </pre>
        </td>
        <td
          class="css-vl0irh-content css-jtsnpx-diff-removed"
        >
          <pre
            class="css-69s43g-content-text"
          >
            <span
              class="css-cncyp1-word-diff"
            >
              <pre
                class="sc-gsTEea bovuND"
              >
                        
                <span
                  class="hljs-string"
                >
                  "
                </span>
              </pre>
            </span>
            <span
              class="css-cncyp1-word-diff css-1uzas5h-word-removed"
            >
              <pre
                class="sc-gsTEea bovuND"
              >
                lightGrey
              </pre>
            </span>
            <span
              class="css-cncyp1-word-diff"
            >
              <pre
                class="sc-gsTEea bovuND"
              >
                <span
                  class="hljs-string"
                >
                  ": "
                </span>
                #
              </pre>
            </span>
            <span
              class="css-cncyp1-word-diff css-1uzas5h-word-removed"
            >
              <pre
                class="sc-gsTEea bovuND"
              >
                BDC0C7
              </pre>
            </span>
            <span
              class="css-cncyp1-word-diff"
            >
              <pre
                class="sc-gsTEea bovuND"
              >
                <span
                  class="hljs-string"
                >
                  ",
                </span>
              </pre>
            </span>
          </pre>
        </td>
      </tr>
      <tr
        class="css-1n7ec1i-line"
      >
        <td
          class="css-jncl07-gutter css-10oj6k6-empty-gutter css-5x3b4c-diff-added"
        >
          <pre
            class="css-na97mt-line-number"
          />
        </td>
        <td
          class="css-jncl07-gutter css-5x3b4c-diff-added"
        >
          <pre
            class="css-na97mt-line-number"
          >
            9
          </pre>
        </td>
        <td
          class="css-1tuzevz-marker css-5x3b4c-diff-added"
        >
          <pre>
            +
          </pre>
        </td>
        <td
          class="css-vl0irh-content css-5x3b4c-diff-added"
        >
          <pre
            class="css-69s43g-content-text"
          >
            <span
              class="css-cncyp1-word-diff"
            >
              <pre
                class="sc-gsTEea bovuND"
              >
                        
                <span
                  class="hljs-string"
                >
                  "
                </span>
              </pre>
            </span>
            <span
              class="css-cncyp1-word-diff css-fjxrmw-word-added"
            >
              <pre
                class="sc-gsTEea bovuND"
              >
                grey
              </pre>
            </span>
            <span
              class="css-cncyp1-word-diff"
            >
              <pre
                class="sc-gsTEea bovuND"
              >
                <span
                  class="hljs-string"
                >
                  ": "
                </span>
                #
              </pre>
            </span>
            <span
              class="css-cncyp1-word-diff css-fjxrmw-word-added"
            >
              <pre
                class="sc-gsTEea bovuND"
              >
                A1A6B0
              </pre>
            </span>
            <span
              class="css-cncyp1-word-diff"
            >
              <pre
                class="sc-gsTEea bovuND"
              >
                <span
                  class="hljs-string"
                >
                  ",
                </span>
              </pre>
            </span>
          </pre>
        </td>
      </tr>
      <tr
        class="css-1n7ec1i-line"
      >
        <td
          class="css-jncl07-gutter css-jtsnpx-diff-removed"
        >
          <pre
            class="css-na97mt-line-number"
          >
            10
          </pre>
        </td>
        <td
          class="css-jncl07-gutter css-10oj6k6-empty-gutter css-jtsnpx-diff-removed"
        >
          <pre
            class="css-na97mt-line-number"
          />
        </td>
        <td
          class="css-1tuzevz-marker css-jtsnpx-diff-removed"
        >
          <pre>
            -
          </pre>
        </td>
        <td
          class="css-vl0irh-content css-jtsnpx-diff-removed"
        >
          <pre
            class="css-69s43g-content-text"
          >
            <span
              class="css-cncyp1-word-diff"
            >
              <pre
                class="sc-gsTEea bovuND"
              >
                        
                <span
                  class="hljs-string"
                >
                  "
                </span>
              </pre>
            </span>
            <span
              class="css-cncyp1-word-diff css-1uzas5h-word-removed"
            >
              <pre
                class="sc-gsTEea bovuND"
              >
                grey
              </pre>
            </span>
            <span
              class="css-cncyp1-word-diff"
            >
              <pre
                class="sc-gsTEea bovuND"
              >
                <span
                  class="hljs-string"
                >
                  ": "
                </span>
              </pre>
            </span>
            <span
              class="css-cncyp1-word-diff css-1uzas5h-word-removed"
            >
              <pre
                class="sc-gsTEea bovuND"
              >
                orange
              </pre>
            </span>
            <span
              class="css-cncyp1-word-diff"
            >
              <pre
                class="sc-gsTEea bovuND"
              >
                <span
                  class="hljs-string"
                >
                  "
                </span>
              </pre>
            </span>
          </pre>
        </td>
      </tr>
      <tr
        class="css-1n7ec1i-line"
      >
        <td
          class="css-jncl07-gutter css-10oj6k6-empty-gutter css-5x3b4c-diff-added"
        >
          <pre
            class="css-na97mt-line-number"
          />
        </td>
        <td
          class="css-jncl07-gutter css-5x3b4c-diff-added"
        >
          <pre
            class="css-na97mt-line-number"
          >
            10
          </pre>
        </td>
        <td
          class="css-1tuzevz-marker css-5x3b4c-diff-added"
        >
          <pre>
            +
          </pre>
        </td>
        <td
          class="css-vl0irh-content css-5x3b4c-diff-added"
        >
          <pre
            class="css-69s43g-content-text"
          >
            <span
              class="css-cncyp1-word-diff"
            >
              <pre
                class="sc-gsTEea bovuND"
              >
                        
                <span
                  class="hljs-string"
                >
                  "
                </span>
              </pre>
            </span>
            <span
              class="css-cncyp1-word-diff css-fjxrmw-word-added"
            >
              <pre
                class="sc-gsTEea bovuND"
              >
                lightGrey
              </pre>
            </span>
            <span
              class="css-cncyp1-word-diff"
            >
              <pre
                class="sc-gsTEea bovuND"
              >
                <span
                  class="hljs-string"
                >
                  ": "
                </span>
              </pre>
            </span>
            <span
              class="css-cncyp1-word-diff css-fjxrmw-word-added"
            >
              <pre
                class="sc-gsTEea bovuND"
              >
                #BDC0C7
              </pre>
            </span>
            <span
              class="css-cncyp1-word-diff"
            >
              <pre
                class="sc-gsTEea bovuND"
              >
                <span
                  class="hljs-string"
                >
                  "
                </span>
              </pre>
            </span>
          </pre>
        </td>
      </tr>
      <tr
        class="css-1n7ec1i-line"
      >
        <td
          class="css-jncl07-gutter"
        >
          <pre
            class="css-na97mt-line-number"
          >
            11
          </pre>
        </td>
        <td
          class="css-jncl07-gutter"
        >
          <pre
            class="css-na97mt-line-number"
          >
            11
          </pre>
        </td>
        <td
          class="css-1tuzevz-marker"
        >
          <pre />
        </td>
        <td
          class="css-vl0irh-content"
        >
          <pre
            class="css-69s43g-content-text"
          >
            <pre
              class="sc-gsTEea bovuND"
            >
                    
              <span
                class="hljs-punctuation"
              >
                }
              </span>
              <span
                class="hljs-punctuation"
              >
                ,
              </span>
            </pre>
          </pre>
        </td>
      </tr>
      <tr
        class="css-1n7ec1i-line"
      >
        <td
          class="css-jncl07-gutter"
        >
          <pre
            class="css-na97mt-line-number"
          >
            12
          </pre>
        </td>
        <td
          class="css-jncl07-gutter"
        >
          <pre
            class="css-na97mt-line-number"
          >
            12
          </pre>
        </td>
        <td
          class="css-1tuzevz-marker"
        >
          <pre />
        </td>
        <td
          class="css-vl0irh-content"
        >
          <pre
            class="css-69s43g-content-text"
          >
            <pre
              class="sc-gsTEea bovuND"
            >
                  
              <span
                class="hljs-punctuation"
              >
                }
              </span>
            </pre>
          </pre>
        </td>
      </tr>
      <tr
        class="css-1n7ec1i-line"
      >
        <td
          class="css-jncl07-gutter"
        >
          <pre
            class="css-na97mt-line-number"
          >
            13
          </pre>
        </td>
        <td
          class="css-jncl07-gutter"
        >
          <pre
            class="css-na97mt-line-number"
          >
            13
          </pre>
        </td>
        <td
          class="css-1tuzevz-marker"
        >
          <pre />
        </td>
        <td
          class="css-vl0irh-content"
        >
          <pre
            class="css-69s43g-content-text"
          >
            <pre
              class="sc-gsTEea bovuND"
            >
                
              <span
                class="hljs-punctuation"
              >
                }
              </span>
            </pre>
          </pre>
        </td>
      </tr>
    </tbody>
  </table>
   
</div>
`;
